### PR TITLE
feat(strategy tools): run baseline strategies as proposers over recorded snapshots

### PR DESCRIPTION
**Pull request by Solders-Girdles**
## Summary

Related issue / finding / routed package: #1164 (stage 1 of 3)

First strategy→proposer convergence step of the accepted [adopt-five-role-composition](../blob/main/docs/decisions/adopt-five-role-composition.md) decision: the stateless baseline strategy family can now run as a `Proposer` over recorder-produced `MarketSnapshot` artifacts, instead of the trade-idea lane growing a second proposer brain.

- **`SnapshotStrategyProposer`** (new, `features/strategy_tools/snapshot_proposer.py`): wraps an injected strategy factory onto the `Proposer` contract. Every `decide(...)` input is derived from the snapshot alone — mark window = candle closes, current mark = last close, flat book (`position_state=None`), zero equity (account state is not part of the snapshot contract). A fresh strategy instance is built per symbol series, so state cannot bleed across symbols or snapshots and identical snapshots yield byte-identical ideas (stable `decision_id`/`record_hash`). Strategies are injected structurally via the new `SnapshotDecider` protocol, so **no new cross-slice import edge**: `strategy_tools` still does not import `live_trade`; composition roots (tests today, the `ideas` CLI in stage 2) own strategy construction.
- **`StrategySignalToTradeIdeaAdapter`**: gains an optional injected `TradeIdeaPositionSizingBridge`. When present, mapped ideas carry executable sizing (positive `quantity`/`notional`, estimated `max_loss.amount`) computed against the same offline budget equity `BaselineProposer` uses, sized at the worst permitted entry. Without it (the default, including the existing default-off runtime gate), advisory sizing is byte-for-byte unchanged. Also adds `actor_id(strategy_name)` so a context-free stable proposer id exists; `proposer_id(context)` delegates to it.

Deliberately out of this PR, per the issue's staging: CLI wiring (`ideas propose-*`/`cycle`/`replay`, stage 2) and the stateful strategies (mean-reversion cooldown, regime-switcher detector; stage 3).

## Type of change
- [ ] Refactor
- [x] Feature
- [ ] Bug fix
- [ ] Tests only
- [ ] CI/Docs/Tooling

## Scope & Impact
- Affected areas / components: `src/gpt_trader/features/strategy_tools/` (new `snapshot_proposer.py`; `trade_idea_adapter.py`; `__init__.py` exports) + unit tests
- Any behavior changes? Additive only. The adapter without a sizing bridge produces identical output to before (pinned by `test_default_adapter_sizing_stays_advisory` and the untouched existing adapter suite); the live engine's default-off proposal gate constructs the adapter without a bridge, so runtime behavior is unchanged. New behavior exists only when a composition root opts in.

## Test Plan
- [x] Unit tests added/updated (`test_snapshot_proposer.py`: protocol conformance, eligibility, executable sizing, determinism/`record_hash` stability, non-buy/insufficient-history/empty-series, fresh-instance-per-series, snapshot-only `decide` inputs, naive-`as_of` coercion; `test_trade_idea_adapter.py`: sized vs advisory paths)
- [ ] Integration tests added/updated (n/a — unit-level slice; end-to-end exercised via CLI drive below)
- [ ] Contract tests (if applicable)
- [x] Ran locally: full `tests/unit` — 6349 passed, 10 skipped
- [x] Markers used appropriately (`unit`/`integration`/`slow`/`perf`)

## Quality Gates
- [x] `make ci-required` passes locally (lint/format, docs audits, type check, agent freshness, test guardrails, core unit tests)
- [x] `uv run mypy src/gpt_trader` clean
- [x] No `sys.path` hacks in tests; `AsyncMock` (not `MagicMock`) on `async def`
- [x] Import boundaries respected (`scripts/ci/check_import_boundaries.py` green; no new edges needed)

## Observability & Errors
- [x] Structured JSON logs for new/changed paths (n/a — the wrapper adds no I/O; failures surface as typed `ValidationError`s from the existing adapter/eligibility paths)
- [x] Errors include diagnostic context (symbol, field, value via `ValidationError`; sub-cent precision guard verified end-to-end)

## Agent Artifacts & Config (if touched)
- [x] `make ci-required` includes the agent-artifact freshness check — green
- [ ] If docs changed: `make agent-docs-links` passes (no docs changed)

## Breaking Changes
- [x] None
- [ ] Documented in PR body and the linked issue if any

## Screenshots / Logs (optional)
End-to-end drive (proposer → `TradeIdeaService` → shipped CLI), golden-cross snapshot:

```
proposer_id: snapshot-strategy-baseline-spot
{ "decision_id": "trade-20260703-baseline-spot-btc-usd-171c9ec6",
  "instrument": "BTC-USD", "direction": "long",
  "entry_zone": ["102.96", "105.04"], "invalidation": "Close below the strategy stop level 101.92",
  "quantity": "0.47600914", "notional": "50.00", "max_loss_amount": "1.49" }
workflow states: ['proposed']
$ gpt-trader ideas approve trade-20260703-... --actor rj --account attested via `ideas budget set`
✓ ideas approve OK (trade-20260703-baseline-spot-btc-usd-171c9ec6, state=approved)
✓ ideas audit verify OK (2 events)
# identical rerun over the same snapshot:
DuplicateTradeIdeaError: Trade idea decision_id 'trade-20260703-...' already exists
```

## Checklist
- [x] Acceptance criteria in linked issue(s), finding packet, or routed package met (stage-1 subset: `Proposer` conformance, eligibility, determinism, executable sizing, no look-ahead, live path unchanged; CLI selectability is stage 2)
- [x] Affected paths listed in PR description

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01RqeG8W2GjNxMk2Wo6agKBa

---
_Generated by [Claude Code](https://claude.ai/code/session_01RqeG8W2GjNxMk2Wo6agKBa)_

## Summary by CodeRabbit

* **New Features**
  * Added snapshot-based strategy proposal to turn recorded market snapshots into executable trade ideas.
  * Enhanced trade-idea sizing: when sizing support is available, generated ideas include executable quantity/notional and computed max-loss; otherwise sizing remains advisory.
  * Exposed snapshot strategy proposer components publicly, including stable proposer IDs and normalized strategy naming.
* **Tests**
  * Added unit tests for snapshot proposer determinism, eligibility mapping, series skipping/insufficient history handling, per-symbol strategy instantiation, snapshot-derived decision inputs, and UTC handling for naive timestamps.

### CODE REVIEW

File: src/gpt_trader/features/strategy_tools/snapshot_proposer.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        self,
+        strategy_factory: StrategyFactory,
+        *,
+        strategy_name: str,
+        adapter: StrategySignalToTradeIdeaAdapter | None = None,
+    ) -> None:
+        if not strategy_name.strip():
+            raise ValidationError("strategy_name must be non-empty", field="strategy_name")
+        self._strategy_factory = strategy_factory
+        self._strategy_name = strategy_name
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/features/strategy_tools/snapshot_proposer.py]: Fixed in ac16bcf — the stripped name is stored now (the slug path already normalized `proposer_id`, but the raw name leaked into thesis/`data_used` and the decision-id fallback reason), pinned by `test_strategy_name_is_normalized`. Also took the `notional > 0` test tightening and the explicit `is None` fallback from the nitpicks.

Skipping the ApplicationContainer suggestion for the default adapter: resolving app-container defaults from inside a `features/` slice would invert the layering, and `BaselineProposer` sets the in-slice precedent of constructing its default sizing bridge inline. The adapter remains constructor-injectable for composition roots (the stage-2 CLI wiring will inject the budget-backed bridge there).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/gpt_trader/features/strategy_tools/snapshot_proposer.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        decision = strategy.decide(
+            symbol=series.symbol,
+            current_mark=closes[-1],
+            position_state=None,
+            recent_marks=closes,
+            equity=Decimal("0"),
+            product=None,
+            candles=series.candles,
+        )
+        context = StrategySignalContext(
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/features/strategy_tools/snapshot_proposer.py:132]: Addressed in e0cea5d via the first remedy: `product_type` is now an explicit constructor parameter passed through to the context. The default stays `SPOT` — for the baseline family the buy path is identical between `SpotStrategy` and `BaselinePerpsStrategy` (they differ only in SELL handling, and non-buy decisions are dropped), so spot-labeled long entries from the shared signal brain remain honest. Any non-spot value fails closed through the adapter's existing "spot ideas only" validation instead of mislabeling the product (pinned by `test_non_spot_product_type_fails_closed`), matching the live gate's fail-closed `_proposal_product_type` stance. Rejecting perps strategy *classes* structurally isn't possible here without importing `live_trade`, which this slice deliberately avoids.

---
_Generated by [Claude Code](https://claude.ai/code)_
